src/nvim/spell.c

```diff
@@ -1309,22 +1310,28 @@ size_t spell_move_to(win_T *wp, int dir, bool allwords, bool curline, hlf_T *att
                             ? p - buf + (ptrdiff_t)len
                             : p - buf) > wp->w_cursor.col)) {
             bool can_spell;
+
+            char *decor_error = NULL;
+            col = (colnr_T)(p - buf);
+
             if (has_syntax) {
-              col = (int)(p - buf);
-              (void)syn_get_id(wp, lnum, (colnr_T)col,
-                               false, &can_spell, false);
-              if (!can_spell) {
-                attr = HLF_COUNT;
-              }
+              (void)syn_get_id(wp, lnum, col, false, &can_spell, false);
             } else {
-              can_spell = true;
+              can_spell =
+                decor_provider_invoke_spell(wp, lnum, col,
+                                            &decor_error)
+                || (wp->w_s->b_p_spo_flags & SPO_NPBUFFER) == 0;
+            }
```

Comment:
You overwrote my change here which allowed `:syntax` and decor providers both allowed to set spell regions. Here `syntax` gets priority.

Reply:
As this is the same behavior for highlighting, I think this makes sense to keep it here too.
What do you think?

Reply:
No. What do you mean?

We want to be able to support both syntax and extmarks to both provide spell regions.

Reply:
I'll make the change you added then